modules/debugger/adapters/python.py

```diff
@@ -7,14 +7,32 @@ class Python(adapter.Adapter):
 	@property
 	def type(self): return 'python'
 
-	async def start(self, log):
+	async def start(self, log, configuration):
 		use_debugpy = get_debugger_setting('python.experimental_debugpy_adapter', False)
+		use_socket = get_debugger_setting('python.experimental_socket_attach', False)
+
+		if use_socket and configuration.request == "attach":
```

Comment:
I think there is a slight issue with when the socket transport is used. I believe it is supposed to only be used if  `connect` is set or there is a top level `port` and not used when `listen` or `processId` is set. 

[Here is the section in vscode about the cases](https://github.com/microsoft/vscode-python/blob/e7136b1448d27e9a607ca0d8242d4272b48aabdb/src/client/debugger/extension/adapter/factory.ts#L36)

We should probably default `host` to localhost if there is a `port` defined that appears to be what vscode is doing.

Reply:
Thanks for the pointer to the `vscode` implementation. I'll make the suggested changes and update the PR after testing.